Simplified/NYPLSettingsAccountDetailViewController.m

```diff
@@ -213,26 +213,37 @@ - (void)viewDidLoad
   self.view.backgroundColor = [NYPLConfiguration backgroundColor];
   self.tableView.keyboardDismissMode = UIScrollViewKeyboardDismissModeInteractive;
 
+  UIActivityIndicatorView *activityIndicator = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle: UIActivityIndicatorViewStyleGray];
+  activityIndicator.center = CGPointMake(self.view.frame.size.width / 2, self.view.frame.size.height / 2);
+  [self.view addSubview:activityIndicator];
+  [activityIndicator startAnimating];
+  
+  void (^completion)(void) = ^() {
+    dispatch_async(dispatch_get_main_queue(), ^{
+      [activityIndicator removeFromSuperview];
+      [self setupViews];
+      self.hiddenPIN = YES;
+      [self accountDidChange];
+      [self updateShowHidePINState];
+    });
+  };
+  
   if (self.businessLogic.libraryAccount.details != nil) {
-    [self setupViews];
+    if ([self.businessLogic juvenileCardsManagementIsPossible]) {
```

Comment:
wondering if we should remove this `if` and always call `checkCardCreationEligibilityWithCompletion` here.  Technically if somehow the business logic is initialized in a way that `juvenileCardsManagementIsPossible` returns true, here we won't check for eligibility even if we should. It seems like we're relying on something external to this class.

It seems more correct to me to always check on eligibility? If so my previous comment about the activity indicator is irrelevant.